stock_partner_delivery_window/models/stock_picking.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from datetime import timedelta
+
+from odoo import _, api, models
+from odoo.tools.misc import format_datetime
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.onchange("scheduled_date")
+    def _onchange_scheduled_date(self):
+        self.ensure_one()
+        if (
+            not self.partner_id
+            or self.partner_id.delivery_time_preference != "time_windows"
+            or self.picking_type_id.code != "outgoing"
+        ):
+            return
+        p = self.partner_id
+        sec_lead_time = self.company_id.security_lead
```

Comment:
this is added by "sale_stock" you need to change the dependencies of this module then or adapt it.

Reply:
Good catch :+1: